server/src/DBTablesMonitoring.cc

```diff
@@ -2101,8 +2099,16 @@ void DBTablesMonitoring::updateHosts(const HostInfoList &hostInfoList,
 	HostInfoListConstIterator newHostsItr = hostInfoList.begin();
 	for (; newHostsItr != hostInfoList.end(); ++newHostsItr) {
 		const HostInfo &newHostInfo = *newHostsItr;
-		if (currValidHosts.erase(newHostInfo.id) >= 1) {
-			// The host already exits. We have nrothing to do.
+		HostIdHostInfoMapIterator currHostItr = currValidHosts.find(newHostInfo.id);
+		if (currHostItr != currValidHosts.end()) {
+			const HostInfo &existHost = *currHostItr->second;
+			if (existHost.hostName != newHostInfo.hostName){
+				updatedHostInfoList.push_back(newHostInfo);
```

Comment:
I believe we can call updatedHostInfoList.push_back() once.
As a result, the code will be simple.